app/controllers/posts_controller.rb

```diff
@@ -25,9 +25,19 @@ def edit
   # POST /posts
   def create
     @post = Post.new(post_params)
-
     if @post.save
-      redirect_to @post, notice: 'Post was successfully created.'
+      tag_params.split(",").each do |tag_new|
+        allready_exist = false
+        Tag.find_each do |tag_db|
```

Comment:
Z "nie przechodz po calej tabeli" chodzi mi o to, ze metoda `find_each`, ktorej uzyles przechodzi po calej tabeli (wiadomo, w najgorszym wypadku, ale wciaz, zwlaszcza ze robisz to dla kazdego taga osobno, wiec duzo razy):

Po pierwsze, jakbys przeczytal dokladnie dokumentacje do metody, ktorej uzyles (zrodlo: http://apidock.com/rails/ActiveRecord/Batches/ClassMethods/find_each):

```ruby
find_each(options = {}) public
Yields each record that was found by the find options. The find is performed by find_in_batches with a batch size of 1000 (or as specified by the :batch_size option).

Example:

  Person.find_each(:conditions => "age > 21") do |person|
    person.party_all_night!
  end
```

To bys wiedzial, ze mozesz przekazac do niej `options` i nawet masz przyklad.

Po drugie - wyslalam ci link - bardzo wazny do przeczytania, ale znowu nie spojrzales na niego.

Po pierwsze: przeczytaj to co ci wyslalam (tu masz po polsku: http://www.apohllo.pl/guides/active_record_querying.html)

CALE!!! Nie tylko otworz i stwierdz, ze jak sa asocjacje na gorze, to juz wszystko wiesz i nie musisz czytac. TYLKO PRZECZYTAJ CALE.

Po drugie, jak bedziesz pisal metode `find_or_create`, uzyj poprawnie wyszukiwania z bazy, najlepiej tak, zeby od razu wyciagnac tagi, ktorych name zawiera sie w tablicy ktora dostaniesz, a potem uzyc sprytnej operacji na tablicach (sprawdz jakie metody ma Array w Ruby) zeby odjac dwie tablice.

Ewentualnie najpierw mozesz uzyc wyciagania pojedynczo, a potem to poprawimy.

Teoretycznie mozesz uzyc tez find_each, ale sprytniej, z odpowiednimi opcjami, ale wolalabym zebys przyswoil sobie wyciaganie rzeczy z bazy poprawnie. Raczej nie uzywa sie`find_each`.